### PR TITLE
feat(go/tree): support array binary tree

### PR DESCRIPTION
**Pull request by Reanon**
If this PR is related to coding or code translation, please fill out the checklist and paste the console outputs to the PR.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

File: codes/go/chapter_tree/array_binary_tree_test.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	"fmt"
+	"testing"
+
+	. "github.com/krahets/hello-algo/pkg"
+)
+
+func TestArrayBinaryTree(t *testing.T) {
+	// 初始化二叉树
+	// 这里借助了一个从数组直接生成二叉树的函数
+	arr := []any{1, 2, 3, 4, 0, 6, 7, 8, 9, 0, 0, 12, 0, 0, 15}
```
Comment (krahets) [inline codes/go/chapter_tree/array_binary_tree_test.go:17]: Can we replace 0 with nil?

File: codes/go/pkg/tree_node.go [hunk 2]
```diff
@@ -25,6 +25,24 @@ func NewTreeNode(v int) *TreeNode {
...
+//	 /——— 3
+//	|    \——— 6
+//	|        \——— 12
+//
+// ——— 1
+//
+//	\——— 2
+//	   |    /——— 9
+//	    \——— 4
+//	        \——— 8
```
Comment (krahets) [inline codes/go/pkg/tree_node.go:45]: The ArrToTree and TreeToArr should be updated to the new implementation. Please refer to Java Python C++

File: codes/go/chapter_tree/array_binary_tree_test.go (hunk 1 above)
Comment (krahets) [reply codes/go/chapter_tree/array_binary_tree_test.go:17]: If not, I think INT_MAX is a better choice.

Comment (Reanon) [reply codes/go/chapter_tree/array_binary_tree_test.go]: done

File: codes/go/pkg/tree_node.go (hunk 2 above)
Comment (Reanon) [reply codes/go/pkg/tree_node.go:40]: done

**Review by krahets — APPROVED** [review]
Thanks for the PR!

**Pull request merged**